adapter/internal/adapter/adapter.go

```diff
@@ -260,26 +261,23 @@ OUTER:
 // fetch APIs from control plane during the server start up and push them
 // to the router and enforcer components.
 func fetchAPIsOnStartUp(conf *config.Config) {
-	// NOTE: Currently controle plane API does not support multiple labels in the same
-	// request. Hence until that is fixed, we have to make seperate requests.
-	// Checking the envrionments to fetch the APIs from
+	// Populate data from config.
+	serviceURL := conf.ControlPlane.ServiceURL
+	userName := conf.ControlPlane.Username
+	password := conf.ControlPlane.Password
 	envs := conf.ControlPlane.EnvironmentLabels
+	skipSSL := conf.ControlPlane.SkipSSLVerification
+	retryInterval := conf.ControlPlane.RetryInterval
+	truststoreLocation := conf.Adapter.Truststore.Location
+
 	// Create a channel for the byte slice (response from the APIs from control plane)
-	c := make(chan synchronizer.SyncAPIResponse)
-	if len(envs) > 0 {
-		// If the envrionment labels are present, call the controle plane
-		// with label concurrently (ControlPlane API is not supported for mutiple labels yet)
-		logger.LoggerMgw.Debugf("Environments label present: %v", envs)
-		go synchronizer.FetchAPIs(nil, envs, c)
-	} else {
-		// If the environments are not give, fetch the APIs from default envrionment
-		logger.LoggerMgw.Debug("Environments label  NOT present. Hence adding \"default\"")
-		envs = append(envs, "default")
-		go synchronizer.FetchAPIs(nil, nil, c)
-	}
+	c := make(chan sync.SyncAPIResponse)
+
+	// Get API details.
+	adapter.GetAPIs(c, nil, serviceURL, userName, password, envs, skipSSL, truststoreLocation,
+		sync.RuntimeArtifactEndpoint, true)
 
-	// Wait for each environment to return it's result
-	for i := 0; i < len(envs); i++ {
+	for i := 0; i < 1; i++ {
```

Comment:
Default LA flow should get APIs from each environment. Why have we changed this in to a single loop?

Reply:
Previously, APIM returned API details per environment. It was not capable to handle multiple environment labels at a time. But now it is fixed from the APIM side and once we call the runtime-artifact endpoint, it returns all the APIs related to all environments at once. So we don't need to iterate several times to retrieve API details. @VirajSalaka Please correct me if I'm wrong.